### ISSUE TITLE
request: promote omake to a library

### ISSUE THREAD

**Issue by mcandre** [gregschmit/omake#22]
Hi,

I like the idea of using Rust to safely parse makefiles. Can we please get a Rust API published for omake?

That way, we can parse and navigate makefile structures with downstream Rust projects!

**Comment by gregschmit** [comment]
Yeah this is planned. I removed the library portion because it was initially causing a lot of complexity because the underlying implementation is very tied to the command line (e.g., command line arguments, sub-make requires the use of the binary), and I wanted a way to have a library component that doesn't include `clap` as a dependency, otherwise consumers of the library would also need `clap`.

Anyway, it's definitely in the pipeline but the last few months have been a bit chaotic or me in my personal life.

**Comment by gregschmit** [comment]
A library target already exists (`[lib] name = "omake"` in Cargo.toml), so omake is consumable as a Rust crate. Public-API stabilization/documentation will happen as part of the 1.0 effort; closing this since the library itself is published.
